File: template_rust/src/driver/api/domain/repository.rs
package domain

import (
	"context"
	"net/http"
	"net/url"
	"template/driver/api"
	"template/pkg/api/domain/repository"

	"template/util/env"
)

type Repository struct {
	env    *env.Env
	client api.Client
}

func NewRepository(
	env *env.Env,
	client api.Client,
) repository.Client {
	return &Repository{
		env:    env,
		client: client,
	}
}

func (r Repository) Get(ctx context.Context, path string, query url.Values) (*http.Response, error) {
	return nil, nil
}

func (r Repository) Post(ctx context.Context, path string, query url.Values) (*http.Response, error) {
	return nil, nil
}

func (r Repository) Put(ctx context.Context, path string, query url.Values) (*http.Response, error) {
	return nil, nil
}

func (r Repository) Delete(ctx context.Context, path string, query url.Values) (*http.Response, error) {
	return nil, nil
}

File: template_go_gorm/src/pkg/api/domain/repository/repository.go
package repository

import (
	"context"
	"net/http"
	"net/url"
)

type Client interface {
	Get(ctx context.Context, path string, query url.Values) (*http.Response, error)
	Post(ctx context.Context, path string, query url.Values) (*http.Response, error)
	Put(ctx context.Context, path string, query url.Values) (*http.Response, error)
	Delete(ctx context.Context, path string, query url.Values) (*http.Response, error)
}
